src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs

```diff
@@ -127,6 +129,51 @@ public static CommandLineBuilder CancelOnProcessTermination(this CommandLineBuil
             return builder;
         }
 
+        public static CommandLineBuilder AbandonOnRepeatCancellation(
```

Comment:
I wonder if enabling this functionality would make sense as a parameter to `CancelOnProcessTermination` rather than a separate method.

Thoughts, @tmds?

Reply:
Maybe as an optional parameter with a struct/class containing behaviour options. Among them bool abandon and int threshold?

Reply:
Yes, I think we should add it to `CancelOnProcessTermination` to avoid weird interactions like https://github.com/dotnet/command-line-api/pull/1227/files#r605602596.

`CancelOnProcessTermination` gets included by `UseDefaults`. There is no pattern for passing patterns to the default middlewares that are included by `UseDefaults`.

Reply:
Are there tools that you've seen where the force quit behavior would involve more than two CTRL-C presses? I'm wondering if the threshold needs to be configurable.

Reply:
True, okay, we don't need a threshold. But @tmds  has a good point. Should UseDefault include this abandoning behaviour? How would someone use UseDefault *and* abandon on secondary cancel?
Should we maybe also inject a state object into the Invocation that keeps a count of cancellations, so that people could react to a secondary cancel? In hosting the StopAsync method accepts a cancellation token to abort graceful shutdown and just force shutdown the host. We could use such an injected status object to control cancellation.

Reply:
My intuition is that this behavior should be enabled by default so that the path of least resistance for the developer leads to the most consistent experience for end users.

@KathleenDollard?